message/router/middleware/poison_test.go

```diff
@@ -2,21 +2,20 @@ package middleware_test
 
 import (
 	"context"
+	stdErrors "errors"
 	"testing"
 	"time"
 
-	"github.com/ThreeDotsLabs/watermill"
-	"github.com/ThreeDotsLabs/watermill/message/subscriber"
-	"github.com/ThreeDotsLabs/watermill/pubsub/gochannel"
-
 	"github.com/hashicorp/go-multierror"
-
-	"github.com/ThreeDotsLabs/watermill/message"
-
-	"github.com/ThreeDotsLabs/watermill/message/router/middleware"
 	"github.com/pkg/errors"
```

Comment:
I dont see a need for this deprecated one

Remove it, so you can use errors directly without an alias

The call to errors.Wrap can be replaced by fmt.Errorf("%w ...")

errors.New can be used directly from stdlib

Reply:
It looks like errors.Cause is being used in the current implementation, which makes it a bit more challenging to simply remove the alias.

Reply:
From what I had seen before raising my point, errors.Cause is used only in the code, not in the test file I quoted.

But, I have missed something. And I was wrong

There is this indeed.

```go
assert.Equal(t, errFailed, errors.Cause(multierr.WrappedErrors()[1]))
```

this could be rewritten to use maybe

```go
assert.ErrorIs(t, multierr, errFailed)
```